### ISSUE TITLE
🚀 Feature: Notifications (webhook?)

### ISSUE THREAD

**Issue by Hukuma1** [stonith404/pingvin-share#88]
### 🔖 Feature description

The ability to send a notification of certain events. On link open, maybe on download of a file, view, etc. Apprise is the gold-standard since it houses basically any notification system you could possibly have/run/want. FOSS as well. https://github.com/caronc/apprise

### 🎤 Pitch

When X share is opened and/or downloaded, I could receive a notification to my Home Assistant which in turn sends me a push notification to my iPhone. Or you might want a message in Discord sent to you. Or Telegram. Endless possibilities. Apprise also has e-mail notifications so in theory could simplify the current e-mail notification code? (not sure)

**Comment by sebastian-b00** [comment]
Yes, currently there is no stable nodejs package available.
I think [this](https://github.com/caronc/apprise-api)  could work, but it may take more work than you would like to use in developing a feature.

**Comment by Hukuma1** [comment]
Yeah, apologies in advance. I wasn't sure what underlying technologies both use and wasn't aware they were incompatible.

Is it possible to just implement Webhooks? Would that be a much smaller ask? Seems most other services (e.g. Telegram, Home Assistant) would still have support for receiving Webhooks.

**Comment by stonith404** [comment]
Yeah webhooks would be easier to implement I think. Which events should trigger the webhook?

**Comment by stonith404** [comment]
@Hukuma1 Just to understand, would you mind to share why you think it would be nice that you can get notifications in e.g Telegram or Discord? Do you have a specific use case?

**Comment by Hukuma1** [comment]
Oh I just grabbed a popular chat program as an example where to send a notification (channel/DM). For my personal use case I would just like to get a push notification to my phone that someone interacted with my share. Since I already use Home Assistant it can be received by a webhook implementation. I just suggested other services to not only make it about my use case. :P

There's a few core events I would say: open and download. Separate, in case someone only clicked the link but didn't actually download? (not sure if each file in the share needs its own event, that's your call)

Future/bonus: If a share is about to expire, can send an alert? New file alert to your reverse share?

**Comment by stonith404** [comment]
@Hukuma1 Okay, thanks for your explanation. I'm looking into it, but currently I want to prioritize other issues if someone can't wait for this feature, feel free to create a PR :)

**Comment by Hukuma1** [comment]
Absolutely! No rush. Keep up the great work!

**Comment by Zixim** [comment]
+1 for webhook implementation

**Comment by stonith404** [comment]
Could you guys suggest me a list of events that would be nice to have for webhooks? E.g `share.create`

**Comment by Zixim** [comment]
when a file is uploaded or downloaded or deleted to/from a (reverse)share. Basically all events related to file movements.
For me, the file upload to any reverse share would be the most interesting one to have.

**Comment by Hukuma1** [comment]
`share.view` = entered share but did not download anything
`share.download` = downloaded
`share.accessfail` = failed pw login (maybe can be invoked is failed X amount of times, eg 3 wrong passwords)

**Comment by CrazyWolf13** [comment]
> Could you guys suggest me a list of events that would be nice to have for webhooks? E.g `share.create`

+1 For some sort of notification service, at best ntfy.sh integration.

For me the following things would be nice:

share viewed
share downloaded
share uploaded
successful login (with IP)
failed login (with IP)

**Comment by maurxce** [comment]
+1

**Comment by Hukuma1** [comment]
Could `ntfy` work for our needs?

https://www.npmjs.com/package/ntfy

https://docs.ntfy.sh/publish/#webhooks-publish-via-get
